prow/tide/tide.go

```diff
@@ -1163,12 +1190,17 @@ func (c *Controller) presubmitsByPull(sp *subpool) (map[int][]config.Presubmit,
 		}
 	}
 
-	for _, ps := range c.config().Presubmits[sp.org+"/"+sp.repo] {
-		if !ps.ContextRequired() {
-			continue
+	for _, pr := range sp.prs {
+		presubmits, err := c.config().GetPresubmits(c.gc, sp.org+"/"+sp.repo, refGetterFactory(sp.sha), refGetterFactory(string(pr.HeadRefOID)))
```

Comment:
nit: `presubmits` is shadowed here. It looks like this still works properly because the `record` closure captures the correct `presubmits` variable, but it would be a clearer to use a different variable name.